### PR TITLE
Shift CI from travis to github actions

### PR DESCRIPTION
**Pull request by aht007**
Issue: [BOM-3005](https://openedx.atlassian.net/browse/BOM-3005)

Description:
Dropped Travis in favor of GitHub Actions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  run_tests:
+    name: Tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-20.04
+        python-version:
+          - 3.8
+        toxenv: [ django22, quality ]
```
Comment (mraarif) [inline .github/workflows/ci.yml:21]: we should be testing Django 3.2 now

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Install dependencies
+      run: pip install -r requirements/ci.txt
+    - name: Run Tests
+      run: make test
+    - name: Run Coverage
+      if: matrix.python-version == '3.8' && matrix.toxenv=='django22'
+      uses: codecov/codecov-action@v2
+      with:
+        flags: unittests
+        fail_ci_if_error: true
```
Comment (mraarif) [inline .github/workflows/ci.yml:40]: file should end with a blank line

File: .github/workflows/pypi-publish.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        run: pip install -r requirements/pip.txt
+
+      - name: Build package
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_UPLOAD_TOKEN }}
```
Comment (mraarif) [inline .github/workflows/pypi-publish.yml:30]: file should end with a blank line

File: Makefile [hunk 4]
```diff
@@ -14,6 +14,7 @@ test:  ## Run the library test suite
 upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -q pip-tools
 	pip install -q -r requirements/pip_tools.txt
+	pip-compile --upgrade -o requirements/pip.txt requirements/pip.in
 	pip-compile --upgrade -o requirements/pip_tools.txt requirements/pip_tools.in
 	pip-compile --upgrade -o requirements/base.txt requirements/base.in
 	pip-compile --upgrade -o requirements/quality.txt requirements/quality.in
```
Comment (mraarif) [inline Makefile:20]: I don't see an update in this target to upgrade requirements in `ci.txt` file

File: tox.ini [hunk 5]
```diff
@@ -7,7 +7,7 @@ deps =
     -r{toxinidir}/requirements/test.txt
 commands =
     coverage run setup.py test
-    coverage report
+    python -m coverage xml
```
Comment (mraarif) [inline tox.ini:10]: why was this change needed?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aht007) [reply .github/workflows/ci.yml]: Updated the environment

**Review by UsamaSadiq — APPROVED** [review]

File: tox.ini (hunk 5 above)
Comment (aht007) [reply tox.ini:10]: Version 2 of Codecov Action does not generate the xml file which version 1 previously did for us hence this change......
Reference PR by Ned: [PR](https://github.com/edx/XBlock/pull/560)

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  run_tests:
+    name: Tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-20.04
+        python-version:
+          - 3.8
+        toxenv: [django32, quality]
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:21]: `os`, `python-version` and `toxenv` should use same notations i.e. `-` or `[]`.

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    steps:
+      - uses: actions/checkout@v2
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:23]: Every step should have a name.

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      - name: setup python
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:24]:
```suggestion
      - name: Setup Python
```

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install APT Packages
+        run: |
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:29]:
```suggestion
        run: sudo apt-get install libblas-dev liblapack-dev gfortran
```

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+          sudo apt-get install libblas-dev liblapack-dev gfortran
+      - name: Install dependencies
+        run: pip install -r requirements/ci.txt
+      - name: Run Tests
+        run: make test
+      - name: Run Coverage
+        if: matrix.python-version == '3.8' && matrix.toxenv=='django22'
+        uses: codecov/codecov-action@v2
+        with:
+          flags: unittests
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:39]: This flag `unittests` is not needed in our case.
```suggestion
```

File: .github/workflows/pypi-publish.yml [hunk 11]
```diff
@@ -0,0 +1,30 @@
+name: Publish package to PyPI
```
Comment (UsamaSadiq) [inline .github/workflows/pypi-publish.yml:1]:
```suggestion
name: Publish Package to PyPI
```

File: .github/workflows/pypi-publish.yml [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+      - "*"
+
+jobs:
+  push:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: setup python
```
Comment (UsamaSadiq) [inline .github/workflows/pypi-publish.yml:15]:
```suggestion
      - name: Setup Python
```

File: .github/workflows/ci.yml (hunk 6 above)
Comment (UsamaSadiq) [reply .github/workflows/ci.yml:21]: Travis is still using `django22` so we should keep that for now. Dropping Django22 support can be done in a different PR later on.
```suggestion
        toxenv: [django22, quality]
```

File: tox.ini [hunk 13]
```diff
@@ -1,13 +1,13 @@
 [tox]
-envlist = py{38}-django{22,30,31},quality
+envlist = py{38}-django{22,30,31,32},quality
```
Comment (UsamaSadiq) [inline tox.ini:2]: This change shouldn't be needed right now. We can do this in a different PR.
```suggestion
envlist = py{38}-django{22,30,31},quality
```

**Review by mraarif — APPROVED** [review]
